src/services/expenses.service.js

```diff
@@ -0,0 +1,81 @@
+const { Expense } = require('../models/Expense.model');
+const { generateRandomId } = require('../Random');
+
+const getAllExpenses = async () => {
+  return Expense.findAll();
+};
+
+const getById = async (id) => {
+  return Expense.findByPk(id);
+};
+
+const normalizeCategory = (categories) => {
+  return Array.isArray(categories) || !categories ? categories : [categories];
+};
+
+const filterExpensesById = (expenses, id) => {
+  return expenses.filter((exp) => exp.userId === parseInt(id));
+};
+
+const filterExpensesByCategory = (expenses, category) => {
+  return expenses.filter((exp) => category.includes(exp.category));
+};
+
+const filterExpensesByDate = (expenses, from, to) => {
+  return expenses.filter((exp) => {
+    const spentAt = new Date(exp.spentAt);
+
+    return spentAt >= from && spentAt <= to;
+  });
+};
+
+const create = async ({
```

Comment:
It seems like you are generating a random ID for each new expense. This is not a good practice because it can lead to collisions. It would be better to let the database handle the generation of IDs.